tenpy/algorithms/dmrg.py

```diff
@@ -492,22 +481,25 @@ def post_update_local(self, i0, update_data, meas_E_trunc=False, upd_env=[False,
             Description
         """
         E0 = update_data['E0']
-        update_LP, update_RP = upd_env
+        i0 = self.i0
         E_trunc = None
         if meas_E_trunc or E0 is None:
             E_trunc = self.env.full_contraction(i0).real  # uses updated LP/RP (if calculated)
             if E0 is None:
                 E0 = E_trunc
             E_trunc = E_trunc - E0
         # now we can also remove the LP and RP on outer bonds, which we don't need any more
-        if update_RP:  # we move to the left -> delete left LP
-            self.env.del_LP(i0)
-            for o_env in self.ortho_to_envs:
-                o_env.del_LP(i0)
-        if update_LP:  # we move to the right -> delete right RP
-            self.env.del_RP(i0 + 1)  # Always +1, even in single site.
-            for o_env in self.ortho_to_envs:
-                o_env.del_RP(i0 + 1)
+        if self.EffectiveH.length == 2:
+            # TODO: Do we need those for single site DMRG? In infinite case?
```

Comment:
We never really _need_ this but it conserves memory if we delete as many `RP`'s and `LP`'s we don't need as possible, right?

Reply:
Kind of true, but for big 2D systems (i.e. large W), the environment takes the largest amount of memory, and then it really is an issue.